kubemarine/kubernetes/__init__.py

```diff
@@ -106,6 +106,12 @@ def enrich_reconfigure_inventory(cluster: KubernetesCluster) -> None:
         default_merger.merge(cluster.inventory.setdefault('services', {}), utils.deepcopy_yaml(kubeadm_sections))
 
 
+@enrichment(EnrichmentStage.ALL)
+def verify_roles(cluster: KubernetesCluster) -> None:
+    if cluster.make_group_from_roles(['control-plane']).is_empty():
```

Comment:
I'm not sure about creating clusters without worker node, because such cluster looks useless for dev case too. For example such cluster can't be installed by default via kubemarine (it fails on calico plugin installation because of pending coredns nodes). 
I can not imagine, in such case only control-plane should be needed. For dev cases it's better to use simple all-in-one cluster.

Reply:
Thank you for feedback.
`verify_roles` checks presence of at least one CP, so no problem here. I suppose you mean potentially adding check of at least one worker.
Single CP cluster might still be installed, see **TestCase 2**, but I agree that it is pretty much useless for dev aims.
I see more useful for dev aims **TestCase 1** which may temporarily leave cluster on one CP.